appengine/common/storage.js

```diff
@@ -60,71 +60,112 @@ BlocklyStorage.restoreBlocks = function() {
+      BlocklyStorage.handleLinkResponse_);
 };
 
 /**
  * Retrieve XML text from database using given key.
  * @param {string} key Key to XML, obtained from href.
  */
 BlocklyStorage.retrieveXml = function(key) {
-  BlocklyStorage.makeRequest_('/storage', 'key', key);
+  BlocklyStorage.makeRequest('/storage', 'key=' + encodeURIComponent(key),
+      BlocklyStorage.handleRetrieveXmlResponse_);
 };
 
 /**
- * Global reference to current AJAX request.
- * @type XMLHttpRequest
- * @private
+ * Global reference to current AJAX requests.
+ * @type Object.<string, XMLHttpRequest>
  */
-BlocklyStorage.xhr_ = null;
+BlocklyStorage.xhr_ = {};
 
 /**
  * Fire a new AJAX request.
  * @param {string} url URL to fetch.
- * @param {string} name Name of parameter.
- * @param {string} content Content of parameter.
- * @private
+ * @param {Array.<string>} data Body of data to be sent in request.
+ * @param {?Function=} opt_onSuccess Function to call after request completes
+ *    successfully.
+ * @param {?Function=} opt_onFailure Function to call after request completes
+ *    unsuccessfully. Defaults to BlocklyStorage alert of request status.
+ * @param {string=} [opt_method='POST'] The HTTP request method to use.
+ */
+BlocklyStorage.makeFormRequest =
+    function(form, opt_onSuccess, opt_onFailure, opt_method) {
+  var data = [];
+  for (var i = 0, element; (element = form.elements[i]); i++) {
+    if (element.name) {
+      data[i] = encodeURIComponent(element.name) + '=' +
+          encodeURIComponent(element.value);
+    }
+  }
+  BlocklyStorage.makeRequest(
+      form.action, data.join('&'), opt_onSuccess, opt_onFailure, opt_method);
+};
+
+/**
+ * Fire a new AJAX request.
+ * @param {string} url URL to fetch.
+ * @param {string} data Body of data to be sent in request.
+ * @param {?Function=} opt_onSuccess Function to call after request completes
+ *    successfully.
+ * @param {?Function=} opt_onFailure Function to call after request completes
+ *    unsuccessfully. Defaults to BlocklyStorage alert of request status.
+ * @param {string=} [opt_method='POST'] The HTTP request method to use.
  */
-BlocklyStorage.makeRequest_ = function(url, name, content) {
-  if (BlocklyStorage.xhr_) {
+BlocklyStorage.makeRequest =
+    function(url, data, opt_onSuccess, opt_onFailure, opt_method) {
+  if (BlocklyStorage.xhr_[url]) {
     // AJAX call is in-flight.
-    BlocklyStorage.xhr_.abort();
+    BlocklyStorage.xhr_[url].abort();
   }
-  BlocklyStorage.xhr_ = new XMLHttpRequest();
-  BlocklyStorage.xhr_.name = name;
-  BlocklyStorage.xhr_.onload = BlocklyStorage.handleRequest_;
-  BlocklyStorage.xhr_.open('POST', url);
-  BlocklyStorage.xhr_.setRequestHeader('Content-Type',
-      'application/x-www-form-urlencoded');
-  BlocklyStorage.xhr_.send(name + '=' + encodeURIComponent(content));
+  BlocklyStorage.xhr_[url] = new XMLHttpRequest();
+  BlocklyStorage.xhr_[url].onload = function() {
+    if (this.status === 200) {
```

Comment:
== is shorter.

Reply:
I was putting "===" to appease my IDE warnings about type cohesion. It's technically in the style guide, but its the "new" one (that also applies to ES6).
If strong preference for "==" I can change.